lab04_example/include/lab04_example/VelocityController.h

```diff
@@ -0,0 +1,73 @@
+/* Author: Luke Dennis l.dennis@unsw.edu.au */
+
+#include <ros/ros.h>
+#include <std_msgs/Int8.h>
+#include <geometry_msgs/Twist.h>
+#include <lab04_example/key_control.h>
+#include <lab04_example/KeyToCmdVel.h>
+
+namespace Lab04 {
+    /* Inspiration from:
+     *  https://github.com/wsnewman/ros_class/blob/master/example_ros_class/src/example_ros_class.h
+     *  https://roboticsbackend.com/roscpp-timer-with-ros-publish-data-at-a-fixed-rate/
+     */
+    class VelocityController {
+        public:
+            VelocityController(ros::NodeHandle nh) : nh_(nh) {
+                key_control_sub_ = nh_.subscribe("KeyControl", 10,
+                        &Lab04::VelocityController::keyCommandCallback_, this);
+                twist_pub_ = nh_.advertise<geometry_msgs::Twist>("cmd_vel", 1);
+                cmd_vel_timer_ = nh.createTimer(ros::Duration(1.0 / 10.0),
+                        std::bind(&Lab04::VelocityController::cmdVelCallback_, this));
+                client_ = nh.serviceClient<lab04_example::KeyToCmdVel>("key_to_cmd_vel");
+            }
+        private:
+            ros::NodeHandle nh_;
+
+            ros::Subscriber key_control_sub_;
+            ros::Publisher 	twist_pub_;
+            ros::Timer cmd_vel_timer_;
+            ros::ServiceClient client_;
+
+            KeyControl::key_press current_cmd_{KeyControl::NULL_KEY};
```

Comment:
Comment on what this is doing, C++ familiarity is low

Reply:
default value construction, probably was covered in 2500